typescript/cli/src/config/ism.ts

```diff
@@ -0,0 +1,246 @@
+import { confirm, input, select } from '@inquirer/prompts';
+import { z } from 'zod';
+
+import { ChainMap, ChainName, IsmType } from '@hyperlane-xyz/sdk';
+
+import { errorRed, log, logBlue, logGreen, logRed } from '../../logger.js';
+import { runMultiChainSelectionStep } from '../utils/chains.js';
+import { FileFormat, mergeYamlOrJson, readYamlOrJson } from '../utils/files.js';
+
+import { readChainConfigsIfExists } from './chain.js';
+
+const MultisigIsmConfigSchema = z.object({
+  type: z.union([
+    z.literal(IsmType.MERKLE_ROOT_MULTISIG),
+    z.literal(IsmType.MESSAGE_ID_MULTISIG),
+  ]),
+  threshold: z.number(),
+  validators: z.array(z.string()),
+});
+
+const RoutingIsmConfigSchema: z.ZodSchema<any> = z.lazy(() =>
+  z.object({
+    type: z.literal(IsmType.ROUTING),
+    owner: z.string(),
+    domains: z.record(IsmConfigSchema),
+  }),
+);
+
+const AggregationIsmConfigSchema: z.ZodSchema<any> = z.lazy(() =>
+  z.object({
+    type: z.literal(IsmType.AGGREGATION),
+    modules: z.array(IsmConfigSchema),
+    threshold: z.number(),
+  }),
+);
+
+const TestIsmConfigSchema = z.object({
+  type: z.literal(IsmType.TEST_ISM),
+});
+
+const IsmConfigSchema = z.union([
+  MultisigIsmConfigSchema,
+  RoutingIsmConfigSchema,
+  AggregationIsmConfigSchema,
+  TestIsmConfigSchema,
+]);
+const IsmConfigMapSchema = z.record(IsmConfigSchema);
+export type ZodIsmConfig = z.infer<typeof IsmConfigSchema>;
+export type ZodIsmConfigMap = z.infer<typeof IsmConfigMapSchema>;
+
+export function parseIsmConfig(filePath: string) {
+  const config = readYamlOrJson(filePath);
+  if (!config) throw new Error(`No ISM config found at ${filePath}`);
+  return IsmConfigMapSchema.safeParse(config);
+}
+
+export function readIsmConfig(filePath: string) {
+  const result = parseIsmConfig(filePath);
+  if (!result.success) {
+    const firstIssue = result.error.issues[0];
+    // throw new Error(
+    //   `Invalid ISM config: ${firstIssue.path} => ${firstIssue.message}`,
+    // );
```

Comment:
Are you certain we should allow users to proceed with an empty ism config here instead of throwing?

Reply:
this was meant for testing. We should be throwing an error here.